modules/propertybasedtesting/include/inviwo/propertybasedtesting/testproperty.h

```diff
@@ -0,0 +1,480 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <inviwo/core/network/processornetwork.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/properties/boolcompositeproperty.h>
+
+#include <filesystem>
+#include <fstream>
+
+#include <inviwo/propertybasedtesting/propertybasedtestingmoduledefine.h>
+#include <inviwo/propertybasedtesting/algorithm/propertyanalyzing.h>
+#include <inviwo/propertybasedtesting/testresult.h>
+
+namespace inviwo {
+
+namespace pbt {
+
+template<typename T>
+class TestPropertyFactoryObjectTemplate;
+
+class IVW_MODULE_PROPERTYBASEDTESTING_API TestPropertyObservable;
+
+class IVW_MODULE_PROPERTYBASEDTESTING_API TestPropertyObserver : public Observer {
+public:
+    friend TestPropertyObservable;
+    TestPropertyObserver() = default;
+    virtual void onTestPropertyChange() {}
+    virtual ~TestPropertyObserver() = default;
+};
+
+class IVW_MODULE_PROPERTYBASEDTESTING_API TestPropertyObservable : public Observable<TestPropertyObserver> {
+protected:
+    TestPropertyObservable() = default;
+    void notifyObserversAboutChange() {
+        forEachObserver([](TestPropertyObserver* o) { o->onTestPropertyChange(); });
+    }
+    virtual ~TestPropertyObservable() = default;
+};
+
+/** \docpage{org.inviwo.PropertyAnalyzer, PropertyAnalyzer}
+ * ![](org.inviwo.NetworkPath.png?classIdentifier=org.inviwo.NetworkPath)
+ *
+ * A NetworkPath<T> is a reference to a Processor or Property of type T in the
+ * ProcessorNetwork. If the pointer to the object is not known (i.e.
+ * after deserialization) or one wants to deserialize a NetworkPath, 
+ * the corresponding ProcessorNetwork must be provided via
+ * setNetwork(...) before the deserialization/first access.
+ */
+template <typename T>
+class IVW_MODULE_PROPERTYBASEDTESTING_API NetworkPath : public Serializable {
```

Comment:
One can't export templates, neither functiosn or classes, hence there should not be any IVW_MODULE_X_API macros here...